src/coreclr/src/pal/src/configure.cmake

```diff
@@ -824,23 +824,25 @@ check_cxx_source_runs("
 int main(void) {
   double pi = 3.14159265358979323846;
   double result;
+  double y = 0.0;
+  double x = 0.0;
 
-  result = atan2(0.0, -0.0);
+  result = atan2(y, -x);
```

Comment:
We should fix the introspection for all of the math methods, not just `atan2`.
I'm also surprised that the compiler doesn't optimize this to the same as well, given that `y` and `x` are essentially constants...

Reply:
Sure, also a suggestion from @Riastradh in Illumos channel; it might be better to use `volatile double y, x` in the introspection code.

Reply:
> I'm also surprised that the compiler doesn't optimize this to the same as well, given that `y` and `x` are essentially constants...

Yes, it is surprising in more than one way. 🙂
The __builtin_atan2 vs. libm's atan2: libm's atan2 has a bug on Illumos, gcc uses its builtin implementation in hard-coded case (as well as when `-O2` optimization flag is supplied, for both hard- and soft- coded) and inlines it; without even calling the libm implementation. Diff of hard vs. soft without -O2: https://paste2.org/bgAHhvcp. In case of clang9, it always calls the libm for both hard- and soft- coded cases, with and without -O2. The libm's bug is that we get 0 instead of PI  with `atan2(0.0, -0.0)`, agnostic of constant'ness.

Reply:
Updated introspection cases related to `math.h` in a58e586; where constant values are passed to functions. Took before/after diffs of cmake configure step on Ubuntu, macOS and SmartOS to ensure there are no (success vs. failed) diffs.